### PR TITLE
chore: remove old signer and refactor tests

### PR DESCRIPTION
**Pull request by cmwaters**
This replaces https://github.com/celestiaorg/celestia-app/pull/2238

It tries to consolidate a lot of our testing infrastructure using the new signer instead of the old

Note that with the exception of signer, this PR touches only tests

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

File: pkg/genesis/vesting_accounts_test.go [hunk 1]
```diff
@@ -30,6 +32,7 @@ const (
 	initBalanceForGasFee = 10
 	vestingAmount        = testfactory.BaseAccountDefaultBalance
 	vestingDelayPerTx    = 10 // this is a safe time (in seconds) to wait for a tx to be executed while the vesting period is not over yet
+	testTimeout          = time.Minute
```
Comment (rootulp) [inline pkg/genesis/vesting_accounts_test.go:35]: 👍

File: test/util/blobfactory/test_util.go [hunk 2]
```diff
@@ -2,22 +2,29 @@ package blobfactory
...
 	coretypes "github.com/tendermint/tendermint/types"
 )
 
-func GenerateManyRawSendTxs(txConfig client.TxConfig, count int) []coretypes.Tx {
-	const acc = "signer"
-	kr := testfactory.GenerateKeyring(acc)
-	signer := blobtypes.NewKeyringSigner(kr, acc, "chainid")
+func DefaultTxOpts() []user.TxOption {
+	return FeeTxOpts(10_000_000)
+}
```
Comment (rootulp) [inline test/util/blobfactory/test_util.go:14]: 👍

File: test/util/malicious/app.go [hunk 3]
```diff
@@ -44,7 +44,7 @@ func (a *App) PrepareProposalHandlerMap() map[string]PrepareProposalHandler {
 type App struct {
 	*app.App
 	maliciousStartHeight      int64
-	malPreparePropsoalHandler PrepareProposalHandler
+	malPrepareProposalHandler PrepareProposalHandler
```
Comment (rootulp) [inline test/util/malicious/app.go:47]: 👍

File: pkg/wrapper/nmt_wrapper_test.go [hunk 4]
```diff
@@ -1,4 +1,4 @@
-package wrapper
+package wrapper_test
```
Comment (rootulp) [inline pkg/wrapper/nmt_wrapper_test.go:1]: 👍 I like the `_test` pattern b/c it forces tests to be written against the public API

**Review by rootulp — APPROVED** [review]
👏 great work! I like the move to more `_test` packages

File: pkg/wrapper/nmt_wrapper_test.go (hunk 4 above)
Comment (cmwaters) [reply pkg/wrapper/nmt_wrapper_test.go:1]: totally. It's generally my default. Only when the public functions become very difficult to test would I result to testing the constituent private methods

**Review by evan-forbes — APPROVED** [review]
LGTM, nice cleanup!!

**Comment by cmwaters** [comment]
Do we want to backport this to v1.x?

**Comment by rootulp** [comment]
Strangely this PR removes a test util helper that is causing tests to fail when bumping the v1.x branch to Go 1.21 so I'm inclined to backport this refactor to v1.x if possible.

**Pull request merged**